pkg/prowgen/prowgen_konflux.go

```diff
@@ -260,7 +260,11 @@ func GenerateKonfluxServerlessOperator(ctx context.Context, openshiftRelease Rep
 		if err != nil {
 			return err
 		}
-		buildArgs := []string{fmt.Sprintf("VERSION=%s", soMetadata.Project.Version)}
+
+		buildArgs := []string{
+			fmt.Sprintf("VERSION=%s", soMetadata.Project.Version),
+			fmt.Sprintf("OPM_IMAGE=registry.redhat.io/openshift4/ose-operator-registry-rhel9:v%s", soMetadata.Requirements.OcpVersion.Max),
```

Comment:
don't we want to stay on RHEL8 for everything?

Reply:
ose-operator-registry has only the tags up to 4.14 (https://catalog.redhat.com/software/containers/openshift4/ose-operator-registry/5cddd0bed70cc57c44b2e1f3), `ose-operator-registry-rhel9` also has the ones afterwards.

Reply:
I have just run across this doc:https://gitlab.cee.redhat.com/konflux/docs/users/-/blob/main/topics/releasing/preparing-for-release.adoc#user-content-publishing-a-fbc-graph
Especially:

> While all of the following RPA samples specify a target OpenShift version with a tag, the actual version (and tag) [are determined by](https://konflux-ci.dev/architecture/ADR/0026-specifying-ocp-targets-for-fbc.html) the parent image used for the FBC build. In order to ship to the proper OCP version, ensure that you are using the properly tagged registry.redhat.io/openshift4/ose-operator-registry parent image

In my opinion, we will need to build multiple different Index images, one for each supported OpenShift version. So, right now it would be 4.13 - 4.17, each version between `min: XX` and `max: YY`. 
I'm saying it here because we probably need to iterate, rather than using just `max`